Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-218.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: b360e8da30dbfefb90d45de07fc5fc08db618cfc
Containerized:True
Number of Error(s): 28
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:671:10: error: \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99 cannot be overloaded
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:669:10: error: with \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:671:10: error: \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99 cannot be overloaded
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:669:10: error: with \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:671:10: error: \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99 cannot be overloaded
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:669:10: error: with \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.cpp:209:6: error: prototype for \xe2\x80\x98void ws_workunits::WsWuInfo::readWorkunitComponentLogs(MemoryBuffer&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99 does not match any in class \xe2\x80\x98ws_workunits::WsWuInfo\xe2\x80\x99
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:176:10: error: candidate is: void ws_workunits::WsWuInfo::readWorkunitComponentLogs(const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:671:10: error: \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99 cannot be overloaded
/home/centos/smoketest/PR-15872/HPCC-Platform/esp/services/ws_workunits/ws_workunitsHelpers.hpp:669:10: error: with \xe2\x80\x98void ws_workunits::CWsWuFileHelper::createWULogFile(IConstWorkUnit*, ws_workunits::WsWuInfo&, const char*, unsigned int, LogAccessReturnColsMode, LogAccessLogFormat, unsigned int)\xe2\x80\x99
make[
 ... ( comment is too long, 53 lines, 7853 bytes, truncated) ... 
[Build error:](https://gist.github.com/36916f5749abf8ef5622a9848407a1e4)
Build log [Build log](https://gist.github.com/HPCCSmoketest/36916f5749abf8ef5622a9848407a1e4#file-result-22-04-01-02-08-22-log)

```

--- a/esp/scm/ws_logaccess.ecm
+++ b/esp/scm/ws_logaccess.ecm
@@ -21,7 +21,9 @@ ESPenum LogAccessType : int
     ByJobIdID(1, "ByJobIdID"),
     ByComponent(2, "ByComponent"),
     ByLogType(3, "ByLogType"),
-    ByTargetAudience(4, "ByTargetAudience")
+    ByTargetAudience(4, "ByTargetAudience"),
+    BySourceInstance(5, "BySourceInstance"),
+    BySourceNode(6, "BySourceNode")
 };
 
 ESPenum LogAccessLogFormat : int

--- a/esp/services/ws_logaccess/WsLogAccessService.cpp
+++ b/esp/services/ws_logaccess/WsLogAccessService.cpp
@@ -92,6 +92,16 @@ bool Cws_logaccessEx::onGetLogs(IEspContext &context, IEspGetLogsRequest &req, I
             logFetchOptions.setFilter(getAudienceLogAccessFilter(targetAud));
             break;
         }
+        case CLogAccessType_BySourceInstance:
+        {
+            logFetchOptions.setFilter(getInstanceLogAccessFilter(searchByValue));
+            break;
+        }
+        case CLogAccessType_BySourceNode:
+        {
+            logFetchOptions.setFilter(getHostLogAccessFilter(searchByValue));
+            break;
+        }
         case LogAccessType_Undefined:
         default:
             throw makeStringException(-1, "Invalid remote log access request type");

--- a/helm/hpcc/values.schema.json
+++ b/helm/hpcc/values.schema.json
@@ -2617,7 +2617,7 @@
         "type": {
           "type": "string",
           "description" : "The searchable HPCC log column to be mapped - 'global' applies to all known fields",
-          "enum": [ "global", "workunits", "components", "audience", "class" ]
+          "enum": [ "global", "workunits", "components", "audience", "class", "instance", "host"]
         },
         "timeStampColumn": {
           "description" : "Name of timestamp column related to mapped field (only requried for 'global' mapping)",

--- a/helm/managed/logging/elastic/elastic4hpcclogs-hpcc-logaccess.yaml
+++ b/helm/managed/logging/elastic/elastic4hpcclogs-hpcc-logaccess.yaml
@@ -21,3 +21,7 @@ global:
           searchColumn: "hpcc.log.audience"          # Field containing audience information
         - type: "class"                              #Search by log class specific log mapping
           searchColumn: "hpcc.log.class"             # Field containing log class information
+        - type: "instance"                           #Search by log source instance specific mapping
+          searchColumn: "kubernetes.pod.name"        # Field containing source instance information
+        - type: "host"                               #Search by log source host specific mapping
+          searchColumn: "kubernetes.node.hostname"   # Field containing source host information

--- a/system/jlib/jlog.cpp
+++ b/system/jlib/jlog.cpp
@@ -3411,6 +3411,16 @@ ILogAccessFilter * getWildCardLogAccessFilter()
     return new FieldLogAccessFilter("", LOGACCESS_FILTER_wildcard);
 }
 
+ILogAccessFilter * getHostLogAccessFilter(const char * host)
+{
+    return new FieldLogAccessFilter(host, LOGACCESS_FILTER_host);
+}
+
+ILogAccessFilter * getInstanceLogAccessFilter(const char * instancename)
+{
+    return new FieldLogAccessFilter(instancename, LOGACCESS_FILTER_instance);
+}
+
 ILogAccessFilter * getJobIDLogAccessFilter(const char * jobId)
 {
     return new FieldLogAccessFilter(jobId, LOGACCESS_FILTER_jobid);
@@ -3515,6 +3525,10 @@ IRemoteLogAccess &queryRemoteLogAccessor()
                     searchColumn: "hpcc.log.audience"
                     - type: "class"
                     searchColumn: "hpcc.log.class"
+                    - type: "instance"
+                    searchColumn: "kubernetes.pod.name"
+                    - type: "host"
+                    searchColumn: "kubernetes.node.hostname"
                 )!!";
                 Owned<IPropertyTree> testTree = createPTreeFromYAMLString(simulatedGlobalYaml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
                 logAccessPluginConfig.setown(testTree->getPropTree("global/logAccess"));

--- a/system/jlib/jlog.hpp
+++ b/system/jlib/jlog.hpp
@@ -1395,6 +1395,8 @@ typedef enum
     LOGACCESS_FILTER_or,
     LOGACCESS_FILTER_and,
     LOGACCESS_FILTER_wildcard,
+    LOGACCESS_FILTER_instance,
+    LOGACCESS_FILTER_host,
     LOGACCESS_FILTER_unknown
 } LogAccessFilterType;
 
@@ -1410,6 +1412,10 @@ inline const char * logAccessFilterTypeToString(LogAccessFilterType field)
         return "audience";
     case LOGACCESS_FILTER_component:
         return "component";
+    case LOGACCESS_FILTER_instance:
+        return "instance";
+    case LOGACCESS_FILTER_host:
+        return "host";
     case LOGACCESS_FILTER_or:
         return "or";
     case LOGACCESS_FILTER_and:
@@ -1434,6 +1440,10 @@ inline unsigned logAccessFilterTypeFromName(char const * name)
         return LOGACCESS_FILTER_audience;
     if(strieq(name, "component"))
         return LOGACCESS_FILTER_component;
+    if(strieq(name, "instance"))
+        return LOGACCESS_FILTER_instance;
+    if(strieq(name, "host"))
+        return LOGACCESS_FILTER_host;
     if(strieq(name, "or"))
         return LOGACCESS_FILTER_or;
     if(strieq(name, "and"))
@@ -1599,6 +1609,8 @@ interface IRemoteLogAccess : extends IInterface
 
 // Helper functions to construct log access filters
 extern jlib_decl ILogAccessFilter * getLogAccessFilterFromPTree(IPropertyTree * tree);
+extern jlib_decl ILogAccessFilter * getInstanceLogAccessFilter(const char * instancename);
+extern jlib_decl ILogAccessFilter * getHostLogAccessFilter(const char * host);
 extern jlib_decl ILogAccessFilter * getJobIDLogAccessFilter(const char * jobId);
 extern jlib_decl ILogAccessFilter * getComponentLogAccessFilter(const char * component);
 extern jlib_decl ILogAccessFilter * getAudienceLogAccessFilter(MessageAudience audience);

--- a/system/logaccess/ElasticStack/ElasticStackLogAccess.cpp
+++ b/system/logaccess/ElasticStack/ElasticStackLogAccess.cpp
@@ -50,9 +50,9 @@ static constexpr const char * DEFAULT_HPCC_LOG_COMPONENT_COL   = "kubernetes.con
 static constexpr const char * DEFAULT_HPCC_LOG_TYPE_COL        = "hpcc.log.class";
 static constexpr const char * DEFAULT_HPCC_LOG_AUD_COL         = "hpcc.log.audience";
 
-static constexpr const char * LOGMAP_INDEXPATTERN_ATT = "@storename";
-static constexpr const char * LOGMAP_SEARCHCOL_ATT = "@searchcolumn";
-static constexpr const char * LOGMAP_TIMESTAMPCOL_ATT = "@timestampcolumn";
+static constexpr const char * LOGMAP_INDEXPATTERN_ATT = "@storeName";
+static constexpr const char * LOGMAP_SEARCHCOL_ATT = "@searchColumn";
+static constexpr const char * LOGMAP_TIMESTAMPCOL_ATT = "@timeStampColumn";
 
 static constexpr const char * DEFAULT_SCROLL_TIMEOUT = "1m"; //Elastic Time Units (i.e. 1m = 1 minute).
 static constexpr std::size_t  DEFAULT_MAX_RECORDS_PER_FETCH = 100;
@@ -131,6 +131,24 @@ ElasticStackLogAccess::ElasticStackLogAccess(const std::vector<std::string> &hos
                 m_audienceIndexSearchPattern = logMap.queryProp(LOGMAP_INDEXPATTERN_ATT);
             if (logMap.hasProp(LOGMAP_SEARCHCOL_ATT))
                 m_audienceSearchColName = logMap.queryProp(LOGMAP_SEARCHCOL_ATT);
+        }
+        else if (streq(logMapType, "instance"))
+        {
+            if (logMap.hasProp(LOGMAP_INDEXPATTERN_ATT))
+                m_instanceIndexSearchPattern = logMap.queryProp(LOGMAP_INDEXPATTERN_ATT);
+            if (logMap.hasProp(LOGMAP_SEARCHCOL_ATT))
+                m_instanceSearchColName = logMap.queryProp(LOGMAP_SEARCHCOL_ATT);
+        }
+        else if (streq(logMapType, "host"))
+        {
+            if (logMap.hasProp(LOGMAP_INDEXPATTERN_ATT))
+                m_hostIndexSearchPattern = logMap.queryProp(LOGMAP_INDEXPATTERN_ATT);
+            if (logMap.hasProp(LOGMAP_SEARCHCOL_ATT))
+                m_hostSearchColName = logMap.queryProp(LOGMAP_SEARCHCOL_ATT);
+        }
+        else
+        {
+            ERRLOG("Encountered invalid LogAccess field map type: '%s'", logMapType);
         }
     }
 
@@ -512,6 +530,38 @@ void ElasticStackLogAccess::populateQueryStringAndQueryIndex(std::string & query
             }
 
             DBGLOG("%s: Searching '%s' component log entries...", COMPONENT_NAME, queryValue.str() );
+            break;
+        }
+        case LOGACCESS_FILTER_host:
+        {
+            if (!m_hostSearchColName.isEmpty())
+            {
+                queryField = m_hostSearchColName.str();
+                fullTextSearch = false; //found dedicated components column
+            }
+
+            if (!m_hostIndexSearchPattern.isEmpty())
+            {
+                queryIndex = m_hostIndexSearchPattern.str();
+            }
+
+            DBGLOG("%s: Searching log entries by host: '%s'", COMPONENT_NAME, queryValue.str() );
+            break;
+        }
+        case LOGACCESS_FILTER_instance:
+        {
+            if (!m_instanceSearchColName.isEmpty())
+            {
+                queryField = m_instanceSearchColName.str();
+                fullTextSearch = false; //found dedicated components column
+            }
+
+            if (!m_instanceIndexSearchPattern.isEmpty())
+            {
+                queryIndex = m_instanceIndexSearchPattern.str();
+            }
+
+            DBGLOG("%s: Searching log entries by HPCC component instance: '%s'", COMPONENT_NAME, queryValue.str() );
             break;
         }
         case LOGACCESS_FILTER_wildcard:

--- a/system/logaccess/ElasticStack/ElasticStackLogAccess.hpp
+++ b/system/logaccess/ElasticStack/ElasticStackLogAccess.hpp
@@ -62,6 +62,12 @@ private:
     StringBuffer m_classSearchColName;
     StringBuffer m_classIndexSearchPattern;
 
+    StringBuffer m_instanceSearchColName;
+    StringBuffer m_instanceIndexSearchPattern;
+
+    StringBuffer m_hostSearchColName;
+    StringBuffer m_hostIndexSearchPattern;
+
     StringBuffer m_defaultDocType; //default doc type to query
 
     elasticlient::Client m_esClient;

